libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/impl/AWSEmailProvider.java

```diff
@@ -0,0 +1,116 @@
+package com.yahoo.athenz.common.server.notification.impl;
+
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.simpleemail.AmazonSimpleEmailService;
+import com.amazonaws.services.simpleemail.AmazonSimpleEmailServiceClientBuilder;
+import com.amazonaws.services.simpleemail.model.RawMessage;
+import com.amazonaws.services.simpleemail.model.SendRawEmailRequest;
+import com.amazonaws.services.simpleemail.model.SendRawEmailResult;
+import com.yahoo.athenz.common.server.notification.EmailProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+import java.io.ByteArrayOutputStream;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Properties;
+
+import static com.yahoo.athenz.common.server.notification.NotificationServiceConstants.CHARSET_UTF_8;
+import static com.yahoo.athenz.common.server.notification.NotificationServiceConstants.HTML_LOGO_CID_PLACEHOLDER;
+
+public class AWSEmailProvider implements EmailProvider {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AWSEmailProvider.class);
+    private final AmazonSimpleEmailService ses;
+
+    @Override
+    public boolean sendEmail(String subject, String body, boolean status, Collection<String> recipients, String from, byte[] logoImage) {
+        try {
+            Session session = Session.getDefaultInstance(new Properties());
```

Comment:
Since creating MIMEMessage is common across diff email providers, it can be extracted in the EmailProvider implementation as default method

Reply:
Yes I'm planning on doing it in the next couple of pull requests. I just want to implement one more email provider before doing it so I'll know exactly what I can extract from this method and what I should keep.